.github/workflows/apexE2ENew.yml

```diff
@@ -0,0 +1,212 @@
+name: Apex End to End Tests - ESM
+
+on:
+  workflow_run:
+    workflows:
+      - Nightly Build Develop
+      - Test, Build, and Release
+    types:
+      - completed
+
+  workflow_dispatch:
+    inputs:
+      automationBranch:
+        description: 'Set the branch to use for automation tests'
+        required: false
+        default: 'testesm'
+        type: string
+      apexLsp:
+        description: 'Apex LSP'
+        required: false
+        default: true
+        type: boolean
+      apexReplayDebugger:
+        description: 'Apex Replay Debugger'
+        required: false
+        default: true
+        type: boolean
+      debugApexTests:
+        description: 'Debug Apex Tests'
+        required: false
+        default: true
+        type: boolean
+      runApexTests:
+        description: 'Run Apex Tests'
+        required: false
+        default: true
+        type: boolean
+      trailApexReplayDebugger:
+        description: 'Trail Apex Replay Debugger'
+        required: false
+        default: true
+        type: boolean
+      vscodeVersion:
+        description: 'VSCode Version'
+        required: false
+        default: '1.90.2'
+        type: string
+      runId:
+        description: 'Run ID of the workflow run that created the vsixes'
+        required: true
+        type: string
+
+  workflow_call:
+    inputs:
+      automationBranch:
+        description: 'Set the branch to use for automation tests'
+        required: false
+        default: 'testesm'
+        type: string
+      apexLsp:
+        description: 'Apex LSP'
+        required: false
+        default: true
+        type: boolean
+      apexReplayDebugger:
+        description: 'Apex Replay Debugger'
+        required: false
+        default: true
+        type: boolean
+      debugApexTests:
+        description: 'Debug Apex Tests'
+        required: false
+        default: true
+        type: boolean
+      runApexTests:
+        description: 'Run Apex Tests'
+        required: false
+        default: true
+        type: boolean
+      trailApexReplayDebugger:
+        description: 'Trail Apex Replay Debugger'
+        required: false
+        default: true
+        type: boolean
+      vscodeVersion:
+        description: 'VSCode Version'
+        required: false
+        default: '1.90.2'
+        type: string
+      runId:
+        description: 'Run ID of the workflow run that created the vsixes'
+        required: true
+        type: string
+
+jobs:
+  apexLSP:
+    if: ${{ inputs.apexLsp || (github.event_name == 'workflow_run' && github.event.workflow_run.conclusion == 'success') }}
+    uses: ./.github/workflows/runE2ETest.yml
+    secrets: inherit
+    with:
+      automationBranch: ${{ inputs.automationBranch || 'testesm' }}
+      testToRun: 'apexLsp.e2e.ts'
+      vscodeVersion: ${{ inputs.vscodeVersion || '1.90.2' }}
+      runId: ${{ inputs.runId || github.event.workflow_run.id  }}
+      os: '["macos-12", "ubuntu-latest"]'
```

Comment:
After we transition to ESM, are we still forced to use macos-12, or can we safely switch to using the latest version of macos?

Reply:
Oh good catch, I had forgotten about that. I'll test with latest

Reply:
Can we make the new branch run on latest vscode?

Reply:
I couldn't find a label for that and also chatgpt said 
"In GitHub Actions, there isn't a specific `runs-on` label for Visual Studio Code versions like `macos-latest` for operating systems"

Reply:
I made these 2 new runs with macos-latest and everything passed ✅ (the ❌ is because the slack notification failed as in my fork  I don't have what's needed for that workflow)
- [Apex End to End Tests - ESM](https://github.com/CristiCanizales/salesforcedx-vscode/actions/runs/9747753629)
- [E2E Test Suite - ESM](https://github.com/CristiCanizales/salesforcedx-vscode/actions/runs/9747754932)

Reply:
@CristiCanizales Does `stable` not work?
https://github.com/forcedotcom/salesforcedx-vscode/pull/5398/files

Reply:
ohh right! I completely forgot about that (as well), 100%.